### PR TITLE
Add troubleshooting section

### PR DESCRIPTION
**Pull request by sam-mccall**

### CODE REVIEW

File: troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+Sorry if things are going wrong!
+
+## Update clangd
+
+Every release fixes many bugs, works in more scenarios, and provides clearer
+logs for troubleshooting.
+
+Try to [install the latest major-version release][installation] or even a
+pre-release snapshot before you try anything else!
```
Comment (hokein) [inline troubleshooting.md:13]: nit: maybe add the link to the pre-release snapshot?

File: troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+from source or installed it from a package, these should be in the right place.
+If you've moved the binary around, move it back!
+
+## Dealing with crashes
+
+If clangd is exiting unexpectedly, this is probably a crashing bug in clangd.
+Understanding it better will help you work around it, and if you can file an
+informative [bug report][] we'll try to fix it for the next release.
+
+The most helpful crash reports have a log including a (symbolic) stack trace,
```
Comment (hokein) [inline troubleshooting.md:74]: the words explaining an informative bug report inspires that we should use a [template issue](https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository).

File: troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+crash. This is a much more useful/fixable bug report.
+
+### Crashes when opening a file
+
+If clangd triggers as soon as you open a particular file, it's likely a crash
+when parsing or indexing that file (or an included header). Again it's very
+useful to know what's in the file and minimize it as far as possible (especially
+include).
+
+Note that a background-indexing crash can look very similar.
```
Comment (hokein) [inline troubleshooting.md]: nit: I'd also mention crashes introduced by clang-tidy as well, I used to encounter some bug reports.

File: troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+The standard library headers are often found relative to the compiler.
+Try to configure your project with an absolute path to the compiler
+(`/usr/bin/gcc`, not `gcc`).
+The logs (with `-log=verbose`) can tell you the `cc1` command, including which
+paths were searched for the standard library. You can compare this to the output
+of `clang -### <args>`.
+
+If you're using an unusual compiler (e.g. a cross-compiler for a different
+platform) you may want to pass `--query-driver=/path/to/mygcc` to allow clangd
```
Comment (hokein) [inline troubleshooting.md:56]: link the [doc](https://clang.llvm.org/extra/clangd/Configuration.html) for `query-driver`? And we are missing it in our doc site...

File: troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
+# Troubleshooting
+
+Sorry if things are going wrong!
+
```
Comment (hokein) [inline troubleshooting.md:4]: nit: I'd add a TOC for this page.

File: installation.md [hunk 6]
```diff
@@ -7,8 +7,8 @@ To use clangd, you need:
 
 ## Installing clangd
 
-You'll want a **recent** version of clangd: 7.0 was the first really usable
-release, and 8.0 is much better.
+You'll want a **recent** version of clangd: the current one is 10.0.
+7.0 was the first really usable release, and 8.0 is much better.
```
Comment (hokein) [inline installation.md]: nit: My reading of it is that clangd8 is good enough, just use it, but we should encourage users to use the latest version.

**Review by hokein — APPROVED** [review]
thanks, looks mostly good, just nits.

File: troubleshooting.md (hunk 2 above)
Comment (sam-mccall) [reply troubleshooting.md:74]: Good idea, I'll set one up. Nothing to do here though, I think.

File: troubleshooting.md (hunk 1 above)
Comment (sam-mccall) [reply troubleshooting.md:13]: The installation page now links to prereleases.

File: troubleshooting.md (hunk 4 above)
Comment (sam-mccall) [reply troubleshooting.md:56]: I don't think we'll want to have a separate page for that flag - if there's critical info that's missing here we should add it inline (but I don't think e.g. explaining how globs work is critical, the --help docs are source-of-truth)

File: troubleshooting.md (hunk 3 above)
Comment (sam-mccall) [reply troubleshooting.md]: Done

**Pull request merged**